apps/mobile/src/components/ChatDetailScreen/atoms/index.tsx

```diff
@@ -778,6 +780,23 @@ export const chatMolecule = molecule((getMolecule, getScope) => {
     )
   })
 
+  const tradeOrOriginOfferCurrencyAtom = atom((get) => {
+    const originOffer = get(offerForChatAtom)
+    const tradeChecklistAmountData = getAmountData(
+      get(tradeChecklistAmountAtom)
+    )
+
+    return (
+      tradeChecklistAmountData?.amountData.currency ??
+      originOffer?.offerInfo?.publicPart?.currency ??
+      'USD'
+    )
+  })
+
+  const btcPriceForTradeCurrencyAtom = atom((get) => {
+    return get(createBtcPriceForCurrencyAtom(tradeOrOriginOfferCurrencyAtom))
+  })
```

Comment:
When possible, avoid creating atom inside atom functions

```suggestion
 const btcPriceForTradeCurrencyAtom = createBtcPriceForCurrencyAtom(tradeOrOriginOfferCurrencyAtom)
```

Reply:
ou sh*it thx for noticing!